frontend/pkg/frontend/frontend.go

```diff
@@ -481,25 +488,24 @@ func (f *Frontend) ArmResourceCreateOrUpdate(writer http.ResponseWriter, request
 		// replace any existing tags.
 		if hcpCluster.TrackedResource.Tags != nil {
 			doc.Tags = hcpCluster.TrackedResource.Tags
-			docUpdated = true
 		}
 
-		return docUpdated
+		return true
```

Comment:
Purpose of the return is to indicate whether changes were made and the `DBClient` update function should proceed with the document replacement.  I almost removed this because I don't think there's any cases left in the frontend where we might return false, but there _are_ some cases in the backend.